### PR TITLE
Power Up Placement and Growth

### PR DESCRIPTION
**Pull request by DJSteins**
Places power ups on the ground and grows them linearly from zero to their original size.

### CODE REVIEW

**Comment by DJSteins** [comment]
[Power up scale support](https://trello.com/c/80hLT1Ch/336-power-up-scale-support)

File: Assets/Script/PowerUp/BasePowerUp/Deployable.cs [hunk 1]
```diff
@@ -53,4 +60,28 @@ private void Update()
...
+        Grow();
+    }
+
+    /// <summary>
+    /// Gets the y-coordinate to place the power up at
+    /// </summary>
+    protected virtual float GetYCoord()
+    {
+        // half the height of the object
+        return transform.lossyScale.y / 2.0f;
```
Comment (m4rtint) [inline Assets/Script/PowerUp/BasePowerUp/Deployable.cs]: Why use lossyScale?

Comment (DJSteins) [reply Assets/Script/PowerUp/BasePowerUp/Deployable.cs]: Because it's easy and native unity

File: Assets/Script/PowerUp/GravityWellBehaviour.cs [hunk 2]
```diff
@@ -72,7 +72,15 @@ private void OnTriggerEnter(Collider other)
             if (other.gameObject.CompareTag(StoredStrings.CarTag))
             {
                 CarController carController = other.GetComponent<CarController>();
-                Player possiblePlayer = carController.PlayerInformation.PlayerNumber;
+                Player possiblePlayer;
+                try 
+                {
+                    possiblePlayer = carController.PlayerInformation.PlayerNumber;
```
Comment (m4rtint) [inline Assets/Script/PowerUp/GravityWellBehaviour.cs]: So why you putting try catch here?
 
And can you do a print statement saying what the issue is, call it from 
Warnings class.

Comment (DJSteins) [reply Assets/Script/PowerUp/GravityWellBehaviour.cs]: There was a NullReferenceException being thrown in the console. Probably something to do with PlayerNumber being null. I figured that, if PlayerNumber is null, then it probably isn't a valid player anyways.

Comment (m4rtint) [reply Assets/Script/PowerUp/GravityWellBehaviour.cs]: Alright.

File: Assets/Script/PowerUp/BasePowerUp/Deployable.cs [hunk 3]
```diff
@@ -22,16 +27,18 @@ public override void Activate(PlayerController playerController)
...
         // This puts the powerup behind the car. 
-        Vector3 SpawnPosition = CarTransform.position - m_throwDistance * Rotation; 
-        // TODO: Project the objects onto the ground.
-        transform.position = SpawnPosition;
+        Vector3 SpawnPosition = CarTransform.position - m_throwDistance * Rotation;
+        transform.position = new Vector3(SpawnPosition.x, GetYCoord(), SpawnPosition.z);
+        
+        // make the powerup initially very small
+        m_originalSize = transform.localScale;
+        transform.localScale = new Vector3(0.0f, 0.0f, 0.0f);
```
Comment (m4rtint) [inline Assets/Script/PowerUp/BasePowerUp/Deployable.cs]: There's a Vector3.zero

**Pull request merged**